Source/Initialization/ExternalField.H

```diff
@@ -144,24 +157,58 @@ struct ExternalFieldView
 #if (AMREX_SPACEDIM == 1)
     amrex::Table1D<double> table;
 #elif (AMREX_SPACEDIM == 2)
-    amrex::Table2D<double,amrex::Order::C> table;
+    amrex::Table2D<double> table;
 #elif (AMREX_SPACEDIM == 3)
-    amrex::Table3D<double,amrex::Order::C> table;
+    amrex::Table3D<double> table;
 #endif
+    amrex::IntVect global_size;
 };
 
 class ExternalFieldReader
 {
 public:
-    ExternalFieldReader (std::string const& read_fields_from_path,
-                         std::string const& F_name, std::string const& F_component);
+    ExternalFieldReader (std::string read_fields_from_path,
+                         std::string F_name, std::string F_component,
+                         amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> const& problo,
+                         amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> const& pdx,
+                         amrex::Box const& dombox, bool distributed);
 
-    ExternalFieldView const& getView () const { return m_external_field_view; }
+    [[nodiscard]] ExternalFieldView getView (int li) const;
+
+    [[nodiscard]] ExternalFieldView getView () const;
+
+    void prepare (amrex::BoxArray const& grids,
+                  amrex::DistributionMapping const& dmap,
+                  amrex::IntVect const& ngrow,
+                  std::function<amrex::Real(amrex::Real)> const& get_zlab = nullptr);
+
+    void prepare (amrex::RealBox const& pbox, int moving_dir, int moving_sign,
+                  std::function<amrex::Real(amrex::Real)> const& get_zlab = nullptr);
+
+    [[nodiscard]] bool distributed () const { return m_distributed; }
 
 private:
+
+    void load_data (amrex::RealBox const& pbox);
+    [[nodiscard]] ExternalFieldView make_view (amrex::BaseFab<double> const& fab) const;
+    void make_cache_box (amrex::RealBox const& pbox, int moving_dir, int moving_sign);
+
+    std::string m_file;
+    std::string m_name;
+    std::string m_component;
+    amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> m_problo; // WarpX's initial prob lo
+    amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> m_probdx; // WarpX's cell size
+    amrex::Box m_dombox;                                  // WarpX's initial domain
```

Comment:
This makes them doxygen:
```suggestion
    amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> m_problo; //! WarpX's initial prob lo
    amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> m_probdx; //! WarpX's cell size
    amrex::Box m_dombox;                                  //! WarpX's initial domain
```